### PR TITLE
Spacing

### PR DESCRIPTION
**Pull request by davidpmccormick**
## What is this PR trying to achieve?
Taking responsibility for spacing (margin and padding) away from components, adding root-scope helper classes which operate at each breakpoint.

In the same way that we define `gridClasses`, we can now add `spacingClasses` to an element:

`<div class="{{ {s:2, m:6, l:10} | spacingClasses('margin', 'top') }}">`

will output:

`<div class="margin-top-s2 margin-top-m6 margin-top-l10">`

with the numbers (2, 6, 10) being a multiplier of the `$spacing-unit` (6px) at the given breakpoint (s, m, l).

This should prove to be more scalable than adding margin and padding to component classes. We will potentially have to override these if something needs to be handled differently in a container query, but those instances would definitely be exceptions.

This PR removes spacing concerns from the `row` class. If we're happy with it, the next 'fun' piece of work would be to implement these classes anywhere we currently use `$vertical-spacing-unit`, `$v-spacing-unit` or `$spacing-unit` in the repo. Starts to address #496.

### CODE REVIEW

File: server/views/components/series-container/index.njk [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+<div class="row {{ {s:10, m:10, l:10} | spacingClasses('padding', 'top') }} ">
   <div class="container">
     <div class="grid">
       <div class="{{ {s:4, m:8, l:12, xl: 12} | gridClasses }}">
-        <h2 class="heading--large v-margin-0">{{ model.name }}</h2>
+        <h2 class="heading--large {{ {s:0, m:0, l:0} | spacingClasses('margin', 'top') }} {{ {s:0, m:0, l:0} | spacingClasses('margin', 'bottom') }}">{{ model.name }}</h2>
       </div>
       <div class="{{ {s:4, m:8, l:8} | gridClasses }}">
-        <p class="v-margin-0">{{ model.description }}</p>
+        <p class="{{ {s:0, m:0, l:0} | spacingClasses('margin', 'top') }} {{ {s:0, m:0, l:0} | spacingClasses('margin', 'bottom') }}">{{ model.description }}</p>
```
Comment (jamesgorrie) [inline server/views/components/series-container/index.njk]: Could we not support something like:
```JS
spacingClasses({ margin: ['top', 'bottom'], padding: ['top'] })
```
?

File: client/scss/base/_typography.scss [hunk 2]
```diff
@@ -52,3 +52,7 @@ li {
     margin-bottom: 0;
   }
 }
+
+hr {
```
Comment (davidpmccormick) [inline client/scss/base/_typography.scss]: I take your point. It's a typographic element of sorts, though. Can't think of a better place for it.

File: server/views/components/series-container/index.njk (hunk 1 above)
Comment (davidpmccormick) [inline server/views/components/series-container/index.njk]: I guess that would be cleaner, yes.

File: client/scss/components/_divider.scss [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
 
+.divider--black {
+  background-color: color('black');
+}
+
+.divider--keyline-grey {
+  background-color: color('keyline-grey');
+}
+
+.divider--stub {
```
Comment (davidpmccormick) [inline client/scss/components/_divider.scss]: It's stubby.

**Pull request merged**